src/pnpm.js

```diff
@@ -19,6 +20,32 @@ async function createWorkspacesFile(targetPath, addonInfo, testAppInfo) {
   await fs.writeFile(path.join(targetPath, 'pnpm-workspace.yaml'), content);
 }
 
+/**
+ *
+ * @param {AddonInfo} addonInfo
+ * @param {TestAppInfo} testAppInfo
+ */
+async function handleImperfections(addonInfo, testAppInfo) {
+  /**
+   * https://github.com/pnpm/pnpm/issues/4965#issuecomment-1405264290
+   *
+   * for dependencies in a monorepo, AND when peerDependencies need to be resolved correctly,
+   * the dependencies must use `dependenciesMeta.*.injected: true`,
+   *
+   * more info here:
+   *  https://pnpm.io/package_json#dependenciesmetainjected
+   */
+  let testAppPackageJsonPath = path.join(testAppInfo.location, 'package.json');
+  let testAppPackageJson = await fse.readJSON(testAppPackageJsonPath);
+
+  testAppPackageJson.dependenciesMeta = {
+    [addonInfo.name.dashed]: {
+      injected: true,
+    },
+  };
```

Comment:
This is never written back to the file, isn't it?

Also, we already have a place to extend the file [here](https://github.com/embroider-build/addon-blueprint/blob/main/index.js#L189), maybe we can handle this in one place?